### PR TITLE
feat: title page sequential view

### PR DESCRIPTION
**Pull request by S-Ou**
not merging directly into main - but first into #14 

code not the tidiest - will refactor out but probs at a later stage

### CODE REVIEW

File: src/app/title/page.tsx [hunk 1]
```diff
@@ -209,16 +263,62 @@ const DependencyOrder: React.FC<DependencyOrderProps> = ({ title }) => {
...
+
 export default function Title() {
-	const dummyTitle: DependencyOrderTitleDto = {
+	const [orderType, setOrderType] = useState<"sequential" | "relational">(
+		"relational",
+	);
+
+	function cycleOrderType() {
+		setOrderType(orderType === "sequential" ? "relational" : "sequential");
+	}
```
Comment (pumbas600) [inline src/app/title/page.tsx]: Typically I like to use arrow functions inside components.

```suggestion
	const cycleOrderType = () => {
		setOrderType(orderType === "sequential" ? "relational" : "sequential");
	}
```

File: src/app/title/page.tsx [hunk 2]
```diff
@@ -102,6 +104,58 @@ const DependencyTabPosterContainer = styled.div`
...
+	h3 {
+		font-size: 1.5rem;
+		font-weight: normal;
+		opacity: 0.6;
+	}
+`;
+
+const SequentialAdjacent = styled(SequentialItem).withConfig({
+	shouldForwardProp: (prop) => prop !== "alignRight",
+})<{ alignRight?: boolean }>`
```
Comment (pumbas600) [inline src/app/title/page.tsx]: I'm pretty sure you can simplify this by prefixing the props with `$`
```suggestion
const SequentialAdjacent = styled(SequentialItem)<{ $alignRight?: boolean }>`
```

Comment (pumbas600) [reply src/app/title/page.tsx]: As a sidenote, this alignment doesn't seem to do anything for me (Tested on Chrome)

File: src/app/title/page.tsx [hunk 3]
```diff
@@ -209,16 +263,62 @@ const DependencyOrder: React.FC<DependencyOrderProps> = ({ title }) => {
...
+			<SequentialItem>
+				<LeadPoster src={title.largePosterUrl} alt={title.name} />
+			</SequentialItem>
+			<SequentialAdjacent alignRight={false}>
+				{next && (
+					<>
+						<h2>next</h2>
+						<Poster src={next.smallPosterUrl} alt={next.name} />
+						<h3>{next.releasedAtUtc?.getFullYear()}</h3>
+					</>
```
Comment (pumbas600) [inline src/app/title/page.tsx]: When you shrink the page the lead poster becomes tiny...

image

I would probably use `flex: 1` , `flex: 2`, `flex: 1` for the three posters to ensure the lead poster always remains twice as wide as the side next/prev posters.

**Review by pumbas600 — CHANGES_REQUESTED** [review]
I noticed a few weird things 😓

File: src/app/title/page.tsx (hunk 2 above)
Comment (sou323) [reply src/app/title/page.tsx]: 💀 oh

it seems to still work visually so i'll just leave it as is for now

File: src/app/title/page.tsx (hunk 3 above)
Comment (S-Ou) [reply src/app/title/page.tsx]: yeah, was aware of this one... styling it was a nightmare and i think i'll just circle round to fixing this bug at a later date (plus at this size there's a chance we'll just be adding in some responsive styling anyways)

**Review by pumbas600 — APPROVED** [review]
Sweet, maybe create tickets for fixing those CSS bugs?

**Pull request merged**